common/persistence/visibility/store/elasticsearch/query_interceptors.go

```diff
@@ -66,29 +76,49 @@ func NewValuesInterceptor(
 	namespaceName namespace.Name,
 	saTypeMap searchattribute.NameTypeMap,
 	searchAttributesMapperProvider searchattribute.MapperProvider,
+	nameInterceptor *nameInterceptor,
 ) *valuesInterceptor {
 	return &valuesInterceptor{
 		namespace:                      namespaceName,
 		searchAttributesTypeMap:        saTypeMap,
 		searchAttributesMapperProvider: searchAttributesMapperProvider,
+		nameInterceptor:                nameInterceptor,
 	}
 }
 
 func (ni *nameInterceptor) Name(name string, usage query.FieldNameUsage) (string, error) {
 	fieldName := name
-	if searchattribute.IsMappable(name) {
+	if strings.EqualFold(fieldName, searchattribute.ScheduleID) {
+		if _, err := ni.searchAttributesTypeMap.GetType(fieldName); err != nil {
+			// Not a custom SA, so convert to WorkflowId
+			fieldName = searchattribute.WorkflowID
+		}
+	}
+
+	if searchattribute.IsMappable(fieldName) {
 		mapper, err := ni.searchAttributesMapperProvider.GetMapper(ni.namespace)
 		if err != nil {
 			return "", err
 		}
 		if mapper != nil {
-			fieldName, err = mapper.GetFieldName(name, ni.namespace.String())
+			fieldName, err = mapper.GetFieldName(fieldName, ni.namespace.String())
 			if err != nil {
 				return "", err
 			}
 		}
 	}
 
+	// Store the transformation if the field name has changed
+	if fieldName != name {
+		if ni.fieldTransformations == nil {
+			ni.fieldTransformations = make(map[string]fieldTransformation)
+		}
+		ni.fieldTransformations[fieldName] = fieldTransformation{
+			originalField: name,
+			newField:      fieldName,
+		}
+	}
+
```

Comment:
can we move this block to the same place where we change field name?

Reply:
The field name can also be altered by mapper.GetFieldName. fieldTransformations is meant to be a more generic construct, so I'm storing _any_ field name changes, including those done by mapper.GetFieldName